service/src/main/java/org/cbioportal/genome_nexus/service/annotation/CanonicalTranscriptResolver.java

```diff
@@ -0,0 +1,51 @@
+package org.cbioportal.genome_nexus.service.annotation;
+
+import org.cbioportal.genome_nexus.model.TranscriptConsequence;
+import org.cbioportal.genome_nexus.model.VariantAnnotation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class CanonicalTranscriptResolver
+{
+    private final TranscriptConsequencePrioritizer consequencePrioritizer;
+
+    @Autowired
+    public CanonicalTranscriptResolver(TranscriptConsequencePrioritizer consequencePrioritizer)
+    {
+        this.consequencePrioritizer = consequencePrioritizer;
+    }
+
+    public TranscriptConsequence resolve(VariantAnnotation variantAnnotation)
+    {
+        List<TranscriptConsequence> transcripts = new ArrayList<>();
+
+        for (TranscriptConsequence transcript : variantAnnotation.getTranscriptConsequences())
```

Comment:
I guess this is copied over from the importer but what is canonical differs for mskcc/uniprot/ensembl. Now we are always using canonical ensembl transcript,  if that isn't found the most severe one. Doesn't seem what we want

Reply:
That's right, copied over from importer. With the latest changes to this PR, we are not relying on the canonical transcript anymore when figuring out the hotspots. We may revisit the `CanonicalTranscriptResolver` when we need to use it...